lint/ruff.bzl

```diff
@@ -22,22 +22,34 @@ _MNEMONIC = "ruff"
 def ruff_action(ctx, executable, srcs, config, report, use_exit_code = False):
     """Run ruff as an action under Bazel.
 
+    Ruff will select the configuration file to use for each source file, as documented here:
+    https://docs.astral.sh/ruff/configuration/#config-file-discovery
+
+    Note: all config files are passed to the action.
+    This means that a change to any config file invalidates the action cache entries for ALL
+    ruff actions.
+
+    However this is needed because:
+
+    1. ruff has an `extend` field, so it may need to read more than one config file
+    2. ruff's logic for selecting the appropriate config needs to read the file content to detect
+      a `[tool.ruff]` section.
+
     Args:
         ctx: Bazel Rule or Aspect evaluation context
         executable: label of the the ruff program
         srcs: python files to be linted
-        config: label of the ruff config file (pyproject.toml, ruff.toml, or .ruff.toml)
+        config: labels of ruff config files (pyproject.toml, ruff.toml, or .ruff.toml)
         report: output file to generate
         use_exit_code: whether to fail the build when a lint violation is reported
     """
-    inputs = srcs + [config]
+    inputs = srcs + config
     outputs = [report]
 
     # Wire command-line options, see
     # `ruff help check` to see available options
     args = ctx.actions.args()
     args.add("check")
-    args.add(config, format = "--config=%s")
```

Comment:
Nothing replaced this?

Reply:
Right, instead of explicitly telling it which config to use, we let it look among the configs it can find.

Reply:
I'm not sure this is a correct approach. I think ruff will only be able to find config file in project root, which may not always be the case.